bin/shpec

```diff
@@ -44,97 +54,138 @@ stub_command() {
-
+    print_result "[ '$(sanitize "$2")' != '$(sanitize "$3")' ]" \
+      "Expected [$2] not to equal [$3]"
+  ;;
   xgt )
-    print_result "[[ $2 -gt $3 ]]" "Expected [$2] to be > [$3]";;
-
+    print_result "[ $2 -gt $3 ]" \
+      "Expected [$2] to be > [$3]"
+  ;;
   xlt )
-    print_result "[[ $2 -lt $3 ]]" "Expected [$2] to be < [$3]";;
-
+    print_result "[ $2 -lt $3 ]" \
+      "Expected [$2] to be < [$3]"
+  ;;
   xmatch )
-    print_result "[[ '$2' =~ $3 ]]" "Expected [$2] to match [$3]";;
-
+    print_result "case '$2' in *$3*) :;; *) false;; esac" \
+      "Expected [$2] to match [$3]"
+  ;;
   xno_match )
-    print_result "[[ ! '$2' =~ $3 ]]" "Expected [$2] not to match [$3]";;
-
+    print_result "case '$2' in *$3*) false ;; *) :;; esac" \
+      "Expected [$2] not to match [$3]"
+  ;;
   xpresent )
-    print_result "[[ -n '$2' ]]" "Expected [$2] to be present";;
-
+    print_result "[ -n '$2' ]" \
+      "Expected [$2] to be present"
+  ;;
   xblank )
-    print_result "[[ -z '$2' ]]" "Expected [$2] to be blank";;
+    print_result "[ -z '$2' ]" \
+      "Expected [$2] to be blank"
+  ;;
 
   xfile_present )
-    print_result "[[ -e $2 ]]" "Expected file [$2] to exist";;
-
+    print_result "[ -e $2 ]" \
+      "Expected file [$2] to exist"
+  ;;
   xfile_absent )
-    print_result "[[ ! -e $2 ]]" "Expected file [$2] not to exist";;
-
+    print_result "[ ! -e $2 ]" \
+      "Expected file [$2] not to exist"
+  ;;
   xsymlink )
     link="$(readlink $2)"
-    print_result "[[ '$link' = '$3' ]]" "Expected [$2] to link to [$3], but got [$link]";;
-
+    print_result "[ '$link' = '$3' ]" \
+      "Expected [$2] to link to [$3], but got [$link]"
+  ;;
   xtest )
-    print_result "$2" "Expected $2 to be true";;
-
+    print_result "$2" \
+      "Expected $2 to be true"
+  ;;
   * )
     if type "$1" 2>/dev/null | grep -q 'function'; then
       matcher="$1"; shift
       $matcher "$@"
     else
       print_result false "Error: Unknown matcher [$1]"
-    fi;;
-
+    fi
+  ;;
   esac
 }
 
 print_result() {
   if eval "$1"; then
-    iecho -e "$green$assertion$norm"
+    iecho "$green$assertion$norm"
   else
-    ((failures += 1))
-    iecho -e "$red$assertion"
-    iecho -e "($2)$norm"
+    : $((failures += 1))
+    iecho "$red$assertion"
+    iecho "($2)$norm"
   fi
 }
 
-SHPEC_ROOT=${SHPEC_ROOT:-$([[ -d './shpec' ]] && echo './shpec' || echo '.')}
-
-case "$1" in
-
-  -v|--version )
-
-    echo "$VERSION";;
-
-  * )
-
-    matcher_files="$(find "$SHPEC_ROOT/matchers" -name '*.sh' 2>/dev/null)"
-
-    for matcher_file in $matcher_files; do
-      . "$matcher_file"
-    done
-
-    files="${@:-$(find $SHPEC_ROOT -name '*_shpec.sh')}"
-
-    time for file in $files; do
-      . "$file"
-    done
 
-    [ $failures -eq 0 ] && color=$green || color=$red
-    echo -e "${color}${examples} examples, ${failures} failures${norm}"
+shpec() {
+  (
+  VERSION=0.1.2posix
```

Comment:
Let's keep the version numbers compliant with [semver](http://semver.org/), i.e. just MAJOR.MINOR.PATCH. I think a minor version bump is appropriate here.